src/serialization/interval_log/mod.rs

```diff
@@ -674,61 +683,83 @@ fn system_time_as_fp_seconds(time: time::SystemTime) -> f64 {
     }
 }
 
-named!(start_time<&[u8], LogEntry>,
-    do_parse!(
-        tag!("#[StartTime: ") >>
-        dur: fract_sec_duration >>
-        char!(' ') >>
-        take_until_and_consume!("\n") >>
-        (LogEntry::StartTime(dur))
-));
-
-named!(base_time<&[u8], LogEntry>,
-    do_parse!(
-        tag!("#[BaseTime: ") >>
-        dur: fract_sec_duration >>
-        char!(' ') >>
-        take_until_and_consume!("\n") >>
-        (LogEntry::BaseTime(dur))
-));
-
-named!(interval_hist<&[u8], LogEntry>,
-    do_parse!(
-        tag: opt!(
-            map!(
-                map_res!(
-                    map!(pair!(tag!("Tag="), take_until_and_consume!(",")), |p| p.1),
-                    str::from_utf8),
-                |s| Tag(s))) >>
-        start_timestamp: fract_sec_duration >>
-        char!(',') >>
-        duration: fract_sec_duration >>
-        char!(',') >>
-        max: double >>
-        char!(',') >>
-        encoded_histogram: map_res!(take_until_and_consume!("\n"), str::from_utf8) >>
-        (LogEntry::Interval(IntervalLogHistogram {
-            tag,
-            start_timestamp,
-            duration,
-            max,
-            encoded_histogram
-        }))
-    )
-);
-
-named!(log_entry<&[u8], LogEntry>,
-    alt_complete!(start_time | base_time | interval_hist));
-
-named!(comment_line<&[u8], ()>,
-    do_parse!(tag!("#") >> take_until_and_consume!("\n") >> (()))
-);
-
-named!(legend<&[u8], ()>,
-    do_parse!(tag!("\"StartTimestamp\"") >> take_until_and_consume!("\n") >> (()))
-);
-
-named!(ignored_line<&[u8], ()>, alt!(comment_line | legend));
+fn start_time(input: &[u8]) -> IResult<&[u8], LogEntry, (&[u8], nom::error::ErrorKind)> {
+    map(
+        tuple((
+            tag("#[StartTime: "),
+            fract_sec_duration,
+            char(' '),
+            take_until("\n"),
+            take(1_usize),
+        )),
+        |x| LogEntry::StartTime(x.1),
+    )(input)
+}
+
+fn base_time(input: &[u8]) -> IResult<&[u8], LogEntry, (&[u8], nom::error::ErrorKind)> {
+    map(
+        tuple((
+            tag("#[BaseTime: "),
+            fract_sec_duration,
+            char(' '),
+            take_until("\n"),
+            take(1_usize),
+        )),
+        |x| LogEntry::BaseTime(x.1),
+    )(input)
+}
+
+fn interval_hist(input: &[u8]) -> IResult<&[u8], LogEntry, (&[u8], nom::error::ErrorKind)> {
+    map(
+        tuple((
+            opt(map(
+                map_res(
+                    map(tuple((tag("Tag="), take_until(","), take(1_usize))), |p| {
+                        p.1
+                    }),
+                    str::from_utf8,
+                ),
+                |s| Tag(s),
+            )),
+            fract_sec_duration,
+            char(','),
+            fract_sec_duration,
+            char(','),
+            double,
+            char(','),
+            map_res(take_until("\n"), str::from_utf8),
+            take(1_usize),
```

Comment:
Same with this ­— why `1_usize` specifically?

Reply:
```
error[E0277]: the trait bound `i32: nom::ToUsize` is not satisfied
   --> src/serialization/interval_log/mod.rs:731:33
    |
731 | ...                   take(1)
    |                       ^^^^ the trait `nom::ToUsize` is not implemented for `i32`
```